### ISSUE TITLE
Update from 0.8.3 to 0.8.4 Ubuntu 20.04

### ISSUE THREAD

**Issue by nicedevil007** [openzfs/zfs#10329]
Hey guys,

I recently updated to the last release of Ubuntu 20.04 LTS. Then I figured out that here is a new release as well. So while I was on updating all my stuff I didn't make a stop on ZoL.

Now I'm encountering a few problems:

![grafik](https://user-images.githubusercontent.com/17103076/81906310-5330ed00-95c6-11ea-8f85-17d1d473f446.png)

Kernel is 5.4.15-050415-generic

So I guess I have to remove the old 0.8.3 release and then install 0.8.4 right?
Tryed this with make uninstall commands for the 0.8.3 and then tryed to install 0.8.4, same happens :(

After installing 0.8.4 with those errors I get this output:

```
root@ubuntu:~/ZOL/zfs-0.8.3# zfs --version
zfs-0.8.4-1
zfs-kmod-0.8.3-1
```

After uninstalling the old stuff, something is still in the list:

![grafik](https://user-images.githubusercontent.com/17103076/81906718-f71a9880-95c6-11ea-9de8-01fd15b219a6.png)

What has to be done to get all things right and running?

**Comment by nicedevil007** [comment]
Ok got the first problem fixed by uninstalling all the old stuff with

`dpkg -P`

While I'm using sanoid to create my zfs snapshots I'm encoutering a new problem after installing sanoid. It is depending on ZFS-fuse and this will lead to dependency issues like this:

```
root@ubuntu:~/ZOL/zfs-0.8.4# apt install sanoid -y
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  zfs-fuse
The following NEW packages will be installed:
  sanoid zfs-fuse
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/816 kB of archives.
After this operation, 3,825 kB of additional disk space will be used.
(Reading database ... 123846 files and directories currently installed.)
Preparing to unpack .../zfs-fuse_0.7.0-20_amd64.deb ...
Unpacking zfs-fuse (0.7.0-20) ...
dpkg: error processing archive /var/cache/apt/archives/zfs-fuse_0.7.0-20_amd64.deb (--unpack):
 trying to overwrite '/sbin/zdb', which is also in package zfs 0.8.4-1
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Selecting previously unselected package sanoid.
Preparing to unpack .../sanoid_2.0.3-2_all.deb ...
Unpacking sanoid (2.0.3-2) ...
Errors were encountered while processing:
 /var/cache/apt/archives/zfs-fuse_0.7.0-20_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

I can't find out how to install zfs-fuse_0.8.4.... (I guess there is a version) and tell ubuntu that it shouldn't install the old 0.7.0 package.

**Comment by admnd** [comment]
ZFS/fuse is a very old and now dead ZFS implementation attempted many years ago (in userland) => https://github.com/pscedu/zfs-fuse

**Comment by satmandu** [comment]
The debian/ubuntu packaging for zfs is quite different from the packaging which is created by the deb package creation steps on the wiki here. 

(This is a major annoyance! I had to copy the debian folder from the ubuntu sources and do a bunch of patching to get 0.8.4 working with zsys and the rest of the zfs 20.04 setup.)

The two zfs package sets are **intentionally** mutually incompatible, which WILL create havoc when trying to overwrite one with the other.  You probably want to do a ```sudo apt install -f``` to try to fix the installation issues. If two packages from your created debs are overwriting a file, you may have to force install those debs in sequence.  (Or just wait for a 0.8.4 version of zfs to make it into the ubuntu repositories.)

Having said that, sanoid appears to be requiring the installation of either zfs or zfsutils-linux as per https://github.com/jimsalterjrs/sanoid/blob/master/packages/debian/control .

Once you fix your zfs install, sanoid should just stop complaining, since at that point an installation of zfs should be active on your system.

If things get really screwed up you can screw it up further though by trying to satisfy dependencies using [dummy packages](https://www.pc-freak.net/blog/create-dummy-packages-on-debian-ubuntu-mint-linux-how-to/).

Caveat emptor...

**Comment by johnnyjacq16** [comment]
@nicedevil007 you do not need to install zfs-dkms or zfs-kmod in ubuntu to get zfs modules as zfs is already included into ubuntu 20.04 kernel by default no packages required, just run `modinfo zfs` to see the modules and to use zfs you just need to install zfsutils-linux.

**Comment by andrey42** [comment]
If your ZFS filesystem features list is supported by new package version and you did not add new features unsupported with original Ubuntu 20.04 LTS ZFS packages (for example bookmarks or zstd compression), then you can easily rollback to previous version. 
Take a snapshot of boot and root pools prior to installing new packages, then install new zfs packages, and if you discover smth does not work, load up from Ubuntu 20.04 live cd or resque system(, import pools, then rollback them and export. Did it multiple times without any problems.
